### PR TITLE
Add location to posts

### LINKED ISSUE #33: As a user, I should be able to get nearby posts and set a location on posts.
using geokit-rails

### LINKED ISSUE #43: As a user, I should be able to view local posts sorted by my distance to the posts.

### LINKED ISSUE #44: As a user, I should be able to attach a location/radius to a post.

### PR DESCRIPTION
**Pull request by zhaoalex**
## Changes
- Using IP geocoding in geokit-rails, add location to posts
- Add location when creating new posts
- Only render posts within 5 mile radius
    - Note: we're doing posts as points and fixing a 5 mile radius around the user
    - I believe YikYak did it this way as well, and it makes more sense + is easier to code + is faster
    - Also I think the ordering by distance is pretty slow, yay for scalability
- No tests qq

#### Attached issue?
Resolves #33 
Resolves #43 
Resolves #44

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -11,9 +18,11 @@ def show
   end
 
   def create
-    @post = Post.new(post_params) # TODO: author, location
+    @post = Post.new(post_params) # TODO: author?
```
Comment (ryang72) [inline app/controllers/posts_controller.rb]: posts are anonymous so I think this is okay

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,8 +1,15 @@
 class PostsController < ApplicationController
   # before_action :check_logged_in, only: [:create]
```
Comment (ryang72) [inline app/controllers/posts_controller.rb]: debatable but okay

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,10 +1,13 @@
 <%= render 'shared/header' %>
 <%= link_to 'New post', '/posts/new' %>
 
+<p>Your location: <%= @location.first %>, <%= @location.last %></p>
```
Comment (ryang72) [inline app/views/posts/index.html.erb]: kinda strange but okay, maybe wanna remove or change to current_user distance to this post or something

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -8,6 +8,7 @@
       params: { post_id: @post.id }%>
    spice level: <%= @post.votes_for.size %>
 </h4>
+<h4><%= @post.latitude %>, <%= @post.longitude %></h4>
```
Comment (ryang72) [inline app/views/posts/show.html.erb]: see above

File: config/environments/development.rb [hunk 5]
```diff
@@ -61,4 +61,10 @@
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 
   config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
+
+  class ActionDispatch::Request
+    def remote_ip
+      '23.243.51.30'
```
Comment (ryang72) [inline config/environments/development.rb]: load testing this is kinda sketch

File: Gemfile [hunk 6]
```diff
@@ -29,6 +29,8 @@ gem 'jbuilder', '~> 2.7'
 gem 'bootsnap', '>= 1.4.2', require: false
 
 gem 'devise'
+gem 'geokit-rails'
+
```
[the remark this answers is not in the document]
Comment (zhaoalex) [reply Gemfile]: no

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (zhaoalex) [reply app/views/posts/index.html.erb]: temporary and agreed, will come in later pr

File: config/environments/development.rb (hunk 5 above)
Comment (zhaoalex) [reply config/environments/development.rb]: will figure it out in future pr

**Pull request merged**